ui/v2.5/src/models/list-filter/criteria/criterion.ts

```diff
@@ -358,18 +255,24 @@ export class NumberCriterion extends Criterion {
     return this.value.toString();
   }
 
-  constructor(type: CriterionType, parameterName?: string, options?: number[]) {
-    super();
+  constructor(type: CriterionOption, options?: number[]) {
+    super(type);
 
-    this.type = type;
     this.options = options;
     this.inputType = "number";
+  }
+}
 
-    if (parameterName) {
-      this.parameterName = parameterName;
-    } else {
-      this.parameterName = type;
-    }
+export abstract class ILabeledIdCriterion extends Criterion<ILabeledId[]> {
```

Comment:
While working on #1397 (which I'm rebasing on top of this branch) I'm wondering if it would be possible to move some methods from the specific criterions to this base class. All seem to have `encodeValue` being identical to:
```ts
  public encodeValue() {
    return this.value.map((o) => {
      return encodeILabeledId(o);
    });
  }
```

And also the fields are pretty common:
```ts
  public modifier = GQL.CriterionModifier.IncludesAll;
  public modifierOptions = [
    Criterion.getModifierOption(GQL.CriterionModifier.IncludesAll),
    Criterion.getModifierOption(GQL.CriterionModifier.Includes),
    Criterion.getModifierOption(GQL.CriterionModifier.Excludes),
  ];
  public options: IOptionType[] = [];
  public value: ILabeledId[] = [];
```
with the only difference being in `IncludeAll` being a possible modifier or not (and thus changing the default as well). So this might be added to `ILabeledIdCriterion` as well by giving it a constructor override and passing it an `includeAll` boolean argument to either add it or not. Descendants are already implementing a default / parameterless constructor anyway and calling to the parents constructor to set the `CriterionOption`, so adding a `true` / `false` argument shouldn't be an issue.